protocol-units/execution/opt-executor/src/executor/execution.rs

```diff
@@ -41,51 +41,50 @@ impl Executor {
 			}
 		};
 
+		let block_executor = self.block_executor.clone();
+
 		let block_id = block.block_id.clone();
-		let parent_block_id = {
-			let block_executor = self.block_executor.read().await;
-			block_executor.committed_block_id()
-		};
+		let parent_block_id = block_executor.committed_block_id();
 
 		let mempool_transactions = block.transactions.clone().into_txns();
 
-		let state_compute = {
-			let block_executor = self.block_executor.write().await;
-			block_executor.execute_block(
+		let block_executor_clone = block_executor.clone();
```

Comment:
Twice?

Perhaps it's time to refactor this from the god `&self` object and methods interface that is really forcing either the locking or cloning? #63 for the discussion.

Reply:
I think you're taking a look at an actor pattern implementation that will potentially resolve this. But, the `.clone()` here is fairly low overhead. It's an `Arc` that we're cloning.